### ISSUE TITLE
NATS named client connections

### ISSUE THREAD

**Issue by wkloucek** [owncloud/ocis#6955]
## Is your feature request related to a problem? Please describe.
I'd like our NATS clients to provide a name when connecting to NATS.

Currenlty the NATS client connections are unnamed:

```
~ # nats server report connections

╭────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                    Top 27 Connections out of 27 by subs                                                    │
├─────┬──────────────────────────────┬────────┬─────────┬──────────────┬─────────┬────────┬─────────┬──────────┬──────────┬───────────┬──────┤
│ CID │ Name                         │ Server │ Cluster │ IP           │ Account │ Uptime │ In Msgs │ Out Msgs │ In Bytes │ Out Bytes │ Subs │
├─────┼──────────────────────────────┼────────┼─────────┼──────────────┼─────────┼────────┼─────────┼──────────┼──────────┼───────────┼──────┤
│ 19  │                              │ nats-0 │ nats    │ 10.96.4.29   │         │ 36s    │ 0       │ 0        │ 0 B      │ 0 B       │ 0    │
│ 38  │                              │ nats-1 │ nats    │ 10.96.4.29   │         │ 37s    │ 0       │ 0        │ 0 B      │ 0 B       │ 0    │
│ 37  │                              │ nats-1 │ nats    │ 10.96.7.34   │         │ 37s    │ 0       │ 0        │ 0 B      │ 0 B       │ 0    │
│ 13  │                              │ nats-0 │ nats    │ 10.96.4.17   │         │ 49s    │ 0       │ 0        │ 0 B      │ 0 B       │ 0    │
│ 25  │                              │ nats-0 │ nats    │ 10.96.4.158  │         │ 24s    │ 0       │ 0        │ 0 B      │ 0 B       │ 0    │
│ 22  │                              │ nats-0 │ nats    │ 10.96.7.92   │         │ 28s    │ 0       │ 0        │ 0 B      │ 0 B       │ 0    │
│ 17  │                              │ nats-0 │ nats    │ 10.96.7.145  │         │ 44s    │ 0       │ 0        │ 0 B      │ 0 B       │ 0    │
│ 21  │                              │ nats-0 │ nats    │ 10.96.7.92   │         │ 28s    │ 0       │ 0        │ 0 B      │ 0 B       │ 0    │
│ 27  │ NATS CLI Version development │ nats-0 │ nats    │ 10.96.7.184  │ $SYS    │ 0s     │ 0       │ 0        │ 0 B      │ 0 B       │ 1    │
│ 23  │                              │ nats-0 │ nats    │ 10.96.4.158  │         │ 24s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 31  │                              │ nats-1 │ nats    │ 10.96.7.164  │         │ 51s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 16  │                              │ nats-0 │ nats    │ 10.96.7.145  │         │ 44s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 9   │                              │ nats-0 │ nats    │ 10.96.19.155 │         │ 52s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 24  │                              │ nats-0 │ nats    │ 10.96.4.158  │         │ 24s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 15  │                              │ nats-0 │ nats    │ 10.96.7.145  │         │ 44s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 14  │                              │ nats-0 │ nats    │ 10.96.4.69   │         │ 49s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 9   │                              │ nats-2 │ nats    │ 10.96.16.181 │         │ 27s    │ 3       │ 3        │ 24 B     │ 1.6 KiB   │ 2    │
│ 12  │                              │ nats-2 │ nats    │ 10.96.4.20   │         │ 13s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 18  │                              │ nats-0 │ nats    │ 10.96.4.168  │         │ 41s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 33  │                              │ nats-1 │ nats    │ 10.96.7.173  │         │ 51s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 35  │                              │ nats-1 │ nats    │ 10.96.7.252  │         │ 49s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 36  │                              │ nats-1 │ nats    │ 10.96.4.21   │         │ 44s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 12  │                              │ nats-0 │ nats    │ 10.96.4.22   │         │ 51s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 11  │                              │ nats-0 │ nats    │ 10.96.7.205  │         │ 51s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 39  │                              │ nats-1 │ nats    │ 10.96.7.95   │         │ 35s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 41  │                              │ nats-1 │ nats    │ 10.96.4.212  │         │ 24s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
│ 42  │                              │ nats-1 │ nats    │ 10.96.7.123  │         │ 13s    │ 3       │ 3        │ 24 B     │ 1.5 KiB   │ 2    │
├─────┼──────────────────────────────┼────────┼─────────┼──────────────┼─────────┼────────┼─────────┼──────────┼──────────┼───────────┼──────┤
│     │ Totals for 27 connections    │        │         │              │         │        │ 54      │ 54       │ 432 B    │ 27 KiB    │ 37   │
╰─────┴──────────────────────────────┴────────┴─────────┴──────────────┴─────────┴────────┴─────────┴──────────┴──────────┴───────────┴──────╯

╭────────────────────────────────╮
│     Connections per server     │
├────────┬─────────┬─────────────┤
│ Server │ Cluster │ Connections │
├────────┼─────────┼─────────────┤
│ nats-2 │ nats    │ 2           │
│ nats-1 │ nats    │ 9           │
│ nats-0 │ nats    │ 16          │
╰────────┴─────────┴─────────────╯
```

## Describe the solution you'd like
All NATS client connections are named

## Describe alternatives you've considered
Manually look up IP adresses and map them to services.

## Additional context
oCIS 3.1.0-beta.1

**Comment by kobergj** [comment]
Needs an upstream fix in go-micro, I'll file a PR

**Comment by kobergj** [comment]
go-micro PR: https://github.com/go-micro/plugins/pull/122

I'll use this ticket to unify events stream creation once above PR is merged. Needs a reva PR too...

**Comment by wkloucek** [comment]
@kobergj I found a possible where oCIS NATS connections don't have a name: NATS used as registry :laughing:

But I'll verify when I got time to do so and create another ticket.
